src/Datadog.Trace/HttpOverStreams/DatadogHttpClient.cs

```diff
@@ -57,27 +58,67 @@ private HttpResponse ReadResponse(Stream responseStream)
             const int statusCodeStart = 9;
             const int statusCodeEnd = 12;
             const int startOfReasonPhrase = 13;
+            const int bufferSize = 10;
 
             // TODO: Get this from StringBuilderCache after we determine safe maximum capacity
             var stringBuilder = new StringBuilder();
 
-            var chArray = new byte[1];
-            void GoNextChar()
+            var chArray = new byte[bufferSize];
+
+            async Task GoNextChar()
             {
-                streamPosition++;
-                chArray[0] = (byte)responseStream.ReadByte();
+                await responseStream.ReadAsync(chArray, offset: 0, count: 1).ConfigureAwait(false);
                 currentChar = Encoding.ASCII.GetChars(chArray)[0];
+                streamPosition++;
+            }
+
+            async Task SkipUntil(int requiredStreamPosition)
+            {
+                var advanceCount = requiredStreamPosition - streamPosition;
+                // Not required in release mode, as should only arise from programming error
+                System.Diagnostics.Debug.Assert(advanceCount > 0, "RequiredStreamPosition should be greater than 0");
+                System.Diagnostics.Debug.Assert(advanceCount <= bufferSize, "RequiredStreamPosition should be less than buffer size");
+
+                await responseStream.ReadAsync(chArray, offset: 0, count: advanceCount).ConfigureAwait(false);
+                currentChar = Encoding.ASCII.GetChars(chArray)[advanceCount - 1];
+                streamPosition += advanceCount;
+            }
+
+            async Task ReadUntil(StringBuilder builder, char stopChar)
+            {
+                while (!currentChar.Equals(stopChar))
```

Comment:
What happens if we never find a `stopChar`?

Reply:
For a network stream, we either hang, or the connection is closed when we read over the end, same as the previous implementation 😉